ui/e2e/src/pages/Panel.ts

```diff
@@ -48,17 +53,18 @@ export class Panel {
     this.resizeHandle = this.container.locator('..').locator('.react-resizable-handle');
 
     this.figure = this.container.getByRole('figure');
-    this.loader = this.container.getByLabel('Loading');
+    this.loader = this.container.locator('[aria-label*=Loading]');
   }
 
   async isLoaded() {
     // Wait for the figure to have at least one visible child that is not the loader.
     await expect(async () => {
-      expect(this.figure).not.toContain(this.loader);
+      await expect(this.loader).toHaveCount(0);
 
       const figureChildren = this.figure.locator('*:visible');
-
       expect(await figureChildren.count()).toBeGreaterThan(0);
+
+      await expect(this.loader).toHaveCount(0);
```

Comment:
Checking both before and after looking for children just to be extra sure and avoid a small edge case with timing that might lead to a flaky test. There might be some complex regex or something I could do with a single check, but this was a bit simpler to write.